staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -512,10 +512,20 @@ func (s preparedGenericAPIServer) Run(stopCh <-chan struct{}) error {
 		<-preShutdownHooksHasStoppedCh
 	}()
 
+	// Start the audit backend before any request comes in. This means we must call Backend.Run
+	// before http server start serving. Otherwise the Backend.ProcessEvents call might block.
```

Comment:
see the comment in Line183 of this file
```

	// auditing. The backend is started after the server starts listening.
	AuditBackend audit.Backend
```

Reply:
Line 194 sorry https://github.com/kubernetes/kubernetes/pull/110094/files#diff-68eb390e148644c38174960be4d103c0e377e71eb540aa8c395f7d2278363973R194-R195

Reply:
originally added by https://github.com/kubernetes/kubernetes/pull/45766/files#diff-68eb390e148644c38174960be4d103c0e377e71eb540aa8c395f7d2278363973R279-R286

and then https://github.com/kubernetes/kubernetes/pull/54287 changed the AuditBackend.Run to happen before Serving starts.

So this PR is not changing any behavior, I believe. I updated the comment.

Reply:
I think this is fine - running NonBlockingRun and then Run would cause problems before this change too...